modules/objc/examples/code_snippets/SampleCodeTest.m

```diff
@@ -1368,1143 +1376,1271 @@ - (void) dontTestListenerCertificateAuthenticatorCallback {
 
-// tag::xlistener-config-tls-disable[]
-thisConfig.disableTLS  = true
+    // end::old-listener-config-client-auth-root[]
 
-// end::xlistener-config-tls-disable[]
+    // tag::listener-config-client-auth-self-signed[]
+    thisConfig.authenticator = [[CBLListenerCertificateAuthenticator alloc] initWithBlock: ^BOOL(NSArray * _Nonnull certs) {
+        CFStringRef certCommonName;
+        SecCertificateRef cert = (__bridge SecCertificateRef)(certs[0]);
+        SecCertificateCopyCommonName(cert, &certCommonName);
+        NSString* cn = (NSString*)CFBridgingRelease(certCommonName);
+        
+        if ([_allowlistedCommonNames valueForKey: cn]) {
+            return YES;
+        }
+        return NO;
+    }];
 
-// tag::listener-config-tls-id-nil[]
-thisConfig.tlsIdentity = nil
+    // end::listener-config-client-auth-self-signed[]
+}
 
-// end::listener-config-tls-id-nil[]
+#pragma mark - JSON API
 
+- (void) dontTestToJSONBlob {
+    CBLDatabase* db = self.db;
+    // Example 2. Using Blobs
+    // tag::tojson-blob[]
 
-// tag::old-listener-config-delta-sync[]
-thisConfig.enableDeltaSync = true
+        // Get a document
+        CBLMutableDocument* thisDoc = [[db documentWithID: @"1000"] toMutable];
 
-// end::old-listener-config-delta-sync[]
+        // Get the image and add as a blob to the document
+        NSString* contentType = @"";
+        UIImage* ourImage = [UIImage imageNamed: @"couchbaseimage.png"];
+        NSData* imageData = UIImageJPEGRepresentation(ourImage, 1);
+        CBLBlob* thisBlob = [[CBLBlob alloc] initWithContentType: contentType data: imageData];
+        [thisDoc setBlob: thisBlob forKey: @"avatar"];
 
+        NSString* theBlobAsJSONstringFails = [[thisDoc blobForKey: @"avatar"] toJSON];
 
-// tag::listener-status-check[]
-NSUInteger totalConnections = thisListener.status.connectionCount;
-NSUInteger activeConnections = thisListener.status.activeConnectionCount;
+        // Save blob as part of doc or alternatively as a blob
 
-// end::listener-status-check[]
+        NSError* error = nil;
+        [db saveDocument: thisDoc error: &error];
+        [db saveBlob: thisBlob error: &error]; // <.>
 
+        // Retrieve saved blob as a JSON, reconstitue and check still blob
 
-// tag::old-listener-config-client-auth-root[]
-  // cert is a pre-populated object of type:SecCertificate representing a certificate
-  let rootCertData = SecCertificateCopyData(cert) as Data
-  let rootCert = SecCertificateCreateWithData(kCFAllocatorDefault, rootCertData as CFData)!
-  // Listener:
-  thisConfig.authenticator = ListenerCertificateAuthenticator.init (rootCerts: [rootCert])
+        CBLDocument* sameDoc = [db documentWithID: @"1000"];
+        CBLBlob* sameBlob = [sameDoc blobForKey: @"avatar"];
+        NSString* theBlobAsJSONstring = [sameBlob toJSON];
+        NSDictionary* dict = sameDoc.toDictionary;
+        for (id key in dict) {
+            NSLog(@"Data -- {%@) = {%@}", key, [dict valueForKey: key]);
+        }
```

Comment:
I don't understand the purpose of logging all keys in this example.

Reply:
Honestly, we can entirely drop this toJSON code snippet for blob or just 

1. Get a document from a database.
2. Get a blob object from the document.
3. Print blob as a JSON string.